src/Window.cpp

```diff
@@ -1158,6 +1159,53 @@ QString Window::executeCommand(QString command) {
         ackReset();
         return ACK;
     }
+    else if (function == "getStat") {
+        if (tokens.size() != 2) {
+            return INVALID;
+        }
+        QString stat = tokens.at(1);
+        // Convert stat to a StatsEnum
+        StatsEnum statsEnum;
+        if (stat == "total-distance") {
+            statsEnum = StatsEnum::TOTAL_DISTANCE;
+        }
+        else if (stat == "total-turns") {
+            statsEnum = StatsEnum::TOTAL_TURNS;
+        }
+        else if (stat == "best-run-distance") {
+            statsEnum = StatsEnum::BEST_RUN_DISTANCE;
+        }
+        else if (stat == "best-run-turns") {
+            statsEnum = StatsEnum::BEST_RUN_TURNS;
+        }
+        else if (stat == "current-run-distance") {
+            statsEnum = StatsEnum::CURRENT_RUN_DISTANCE;
+        }
+        else if (stat == "current-run-turns") {
+            statsEnum = StatsEnum::CURRENT_RUN_TURNS;
+        }
+        else if (stat == "total-effective-distance") {
+            statsEnum = StatsEnum::TOTAL_EFFECTIVE_DISTANCE;
+        }
+        else if (stat == "best-run-effective-distance") {
+            statsEnum = StatsEnum::BEST_RUN_EFFECTIVE_DISTANCE;
+        }
+        else if (stat == "current-run-effective-distance") {
+            statsEnum = StatsEnum::CURRENT_RUN_EFFECTIVE_DISTANCE;
+        }
+        else if (stat == "score") {
+            statsEnum = StatsEnum::SCORE;
+        }
+        else {
+            return INVALID;
+        }
+        QString statValue = stats->getStat(statsEnum);
+        if (statValue == "") {
+            // Cannot return an empty string
+            return "-1";
+        }
```

Comment:
I think `-1` is good choice. Can you just add a comment explaining when/why this happens?